pool/app/filter/entity_guard.ml

```diff
@@ -19,12 +19,21 @@ module Access = struct
   open ValidationSet
   open Permission
 
-  let filter action uuid =
-    one_of_tuple (action, `Filter, Some (uuid |> Uuid.target_of Entity.Id.value))
+  let filter ?id ?(model = `Filter) permission =
+    one_of_tuple
+      (permission, model, CCOption.map (Uuid.target_of Entity.Id.value) id)
   ;;
 
   let index = one_of_tuple (Read, `Filter, None)
-  let create = one_of_tuple (Read, `Filter, None)
-  let update = filter Update
-  let delete = filter Delete
+
+  let create ?id () =
```

Comment:
Can we rename this argument `id` to `experiment_id` to make clear, which id to use?